frontend/src/components/MapView/LeftPanel/layersPanel/MenuSwitch/index.tsx

```diff
@@ -89,7 +89,11 @@ const MenuSwitch = memo(({ title, layers, extent }: MenuSwitchProps) => {
   const selectedInternalLayers = useMemo(() => {
     return selectedLayers.filter(layer => {
       return layers.some(internalLayer => {
-        return isEqual(internalLayer, layer);
+        return (
+          layer.id === internalLayer.id ||
+          (internalLayer.group &&
+            internalLayer.group.layers.some(l => l.id === layer.id))
+        );
```

Comment:
Since this is used twice, let's create an explicit function with a nice, telling name please :)

Reply:
Done